api_v3/lib/types/KalturaUserAppRole.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * @package api
+ * @subpackage objects
+ * @relatedService UserAppRoleService
+ */
+
+class KalturaUserAppRole extends KalturaAppRole
+{
+	/**
+	 * @var string
+	 * @insertonly
+	 */
+	public $userId;
+	
+	/**
+	 * @param $kuserToUserRole
+	 * @param $skip
+	 * @return KuserToUserRole|null
+	 *
+	 * @throws KalturaAPIException
+	 * @throws PropelException
+	 * @throws kCoreException
+	 */
+	public function toInsertableObject($kuserToUserRole = null, $skip = array())
+	{
+		if (is_null($kuserToUserRole))
+		{
+			$kuserToUserRole = new KuserToUserRole();
+		}
+		
+		$kuserToUserRole = parent::toInsertableObject($kuserToUserRole, $skip);
+		
+		/* @var KuserToUserRole $kuserToUserRole */
+		$kuserToUserRole->setKuserId($this->kuserId);
+		$kuserToUserRole->setAppGuid($this->appGuid);
+		$kuserToUserRole->setUserRoleId($this->userRoleId);
+		
+		return $kuserToUserRole;
+	}
+	
+	/**
+	 * @throws PropelException
+	 * @throws KalturaAPIException
+	 * @throws kCoreException
+	 */
+	public function validateForInsert($propertiesToSkip = array())
+	{
+		$this->userId = trim($this->userId);
+		$this->appGuid = trim($this->appGuid);
+		$this->userRoleId = trim($this->userRoleId);
+		
+		$this->verifyMandatoryParams($this->userId, $this->appGuid, $this->userRoleId);
+		
+		$partnerId = kCurrentContext::getCurrentPartnerId();
+		
+		// validate userId exists
+		$kuser = kuserPeer::getKuserByPartnerAndUid($partnerId, $this->userId);
+		
+		if (!$kuser)
+		{
+			throw new KalturaAPIException(KalturaErrors::USER_ID_NOT_FOUND, $this->userId);
+		}
+		
+		$this->kuserId = $kuser->getId();
+		KuserToUserRolePeer::isValidForInsert($kuser, $this->appGuid, $this->userRoleId);
+		
+		parent::validateForInsert($propertiesToSkip);
+	}
+	
+	public function toUpdatableObject($kuserToUserRole, $skip = array())
+	{
+		return parent::toUpdatableObject($kuserToUserRole, $skip);
+	}
+	
+	/**
+	 * @throws KalturaAPIException
+	 * @throws kCoreException
+	 */
+	public function validateForUpdate($kuserToUserRole, $propertiesToSkip = array())
+	{
+		KuserToUserRolePeer::isValidForUpdate($this->userRoleId);
+		return parent::validateForUpdate($kuserToUserRole, $propertiesToSkip);
+	}
+	
+	public function doFromObject($kuserToUserRoleObject, KalturaDetachedResponseProfile $responseProfile = null)
+	{
+		/* @var KuserToUserRole $kuserToUserRoleObject*/
+		if(!$kuserToUserRoleObject)
+		{
+			return false;
+		}
+		
+		$this->userId = kuserPeer::retrieveByPK($kuserToUserRoleObject->getKuserId())->getPuserId();
+		
+		parent::doFromObject($kuserToUserRoleObject, $responseProfile);
+	}
+	
+	private function verifyMandatoryParams($puserId, $appGuid ,$userRoleId)
```

Comment:
Maybe try annotating the param with minLength 1 
I think it will remove the need of using this function

Reply:
Was looking into it, not sure it's gonna work
Because for updateAction userId and appGuid should be null (not updatable)

So it's only legible for 'validateForInsert'

Was thinking about 'validateForUsage' but it's being called at "toObject" and will be included for both add and update actions

I think the current function is the more elegant
We can go 'get_class_vars' and make sure non of the properties are null, but don't think it's better.